src/test/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilterUnitTest.java

```diff
@@ -0,0 +1,427 @@
+import org.broadinstitute.hellbender.tools.spark.utils.IntHistogram;
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class XGBoostEvidenceFilterUnitTest extends GATKBaseTest {
+    private static final String testAccuracyDataJsonFile = publicTestDir + "sv_classifier_test_data.json";
+    private static final String classifierModelFile = "/large/sv_evidence_classifier.bin";
+    private static final String resourceClassifierModelFile = "gatk-resources::" + classifierModelFile;
+    private static final String localClassifierModelFile
+            = new File(publicMainResourcesDir, classifierModelFile).getAbsolutePath();
+    private static final String testFeaturesJsonFile = publicTestDir + "sv_features_test_data.json";
+    private static final double probabilityTol = 2.0e-4;
+    private static final double featuresTol = 1.0e-6;
+
+    private final ClassifierAccuracyData classifierAccuracyData = new ClassifierAccuracyData(testAccuracyDataJsonFile);
+    private final double[] predictYProbaSerial = predictProba(
+            XGBoostEvidenceFilter.loadPredictor(localClassifierModelFile), classifierAccuracyData.features
+    );
+    private static final FeaturesTestData featuresTestData = new FeaturesTestData(testFeaturesJsonFile);
+
+    private static final StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection params
+            = new StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection();
+
+    private static final SAMFileHeader artificialSamHeader = initSAMFileHeader();
+    private static final String readGroupName = "Pond-Testing";
+    private static final String DEFAULT_SAMPLE_NAME = "SampleX";
+    public static final ReadMetadata readMetadata = initMetadata();
+    private static final PartitionCrossingChecker emptyCrossingChecker = new PartitionCrossingChecker();
+    private final List<BreakpointEvidence> evidenceList = Arrays.stream(featuresTestData.stringReps)
+            .map(strRep -> BreakpointEvidence.fromStringRep(strRep, readMetadata)).collect(Collectors.toList());
+
+    private static SAMFileHeader initSAMFileHeader() {
+        final SAMFileHeader samHeader = createArtificialSamHeader();
+        SAMReadGroupRecord readGroup = new SAMReadGroupRecord(readGroupName);
+        readGroup.setSample(DEFAULT_SAMPLE_NAME);
+        samHeader.addReadGroup(readGroup);
+        return samHeader;
+    }
+
+    /**
+     * Create synthetic SAM Header comptible with genome tracts (e.g. has all the primary contigs)
+     * Use genome gaps file to enforce this.
+     */
+    public static SAMFileHeader createArtificialSamHeader() {
+        Comparator<String> contigComparator = new Comparator<String>() {
+            @Override public int compare(String s1, String s2) {
```

Comment:
There has to already be some natural ordering comparator for contigs somewhere in the GATK or htsjdk.